### PR TITLE
Added mentions and company models

### PR DESCRIPTION
**Pull request by derekjou**
Modified settings to refer to company model, which was newly added. The company document will reference mention documents.

### CODE REVIEW

**Comment by ncen5293** [comment]
When I make a new account, it makes a user document then crashes the backend and is unable to make a user settings document.
You would also have to change how we get and set companies 
image

**Comment by derekjou** [comment]
> When I make a new account, it makes a user document then crashes the backend and is unable to make a user settings document.
> You would also have to change how we get and set companies
> image

Fixed! I won't be able to integrate the mentions model fully yet until we get the OK from Aecio on the interface. Reddit and Twitter routes will have to be modified for this as well later!

**Review by ncen5293 — COMMENTED** [review]
Everything looks good except you could just merge dev into this branch and in the /:email route just get the companies the same way you did with /:email/company for using the interface and giving the company names to the frontend

File: server/routes/settings.js [hunk 1]
```diff
@@ -48,23 +58,28 @@ router.put("/:email/platform/:platform", jwtVerify, (req, res, next) => {
...
+
+// Gets users settings
+router.get("/:email", jwtVerify, (req, res, next) => {
+  SettingsModel.findOne({ email: req.params.email }).populate('companies').exec((err, settings) => {
     if (settings) {
+      let companyNames = [];
+      settings.companies.forEach(company => companyNames.push(company.name));
       const interface = new Interface();
-      const mentions = await interface.getNewestMentions(settings.companies);
+      const mentions = await interface.getNewestMentions(companyNames);
```
Comment (ncen5293) [inline server/routes/settings.js]: can you also make it an async route since it is using await?
and can you pass the companyNames to the frontend too

**Review by ncen5293 — CHANGES_REQUESTED** [review]

**Pull request merged**